jenkins-plugin/src/main/java/org/jenkinsci/plugins/pipeline/maven/WithMavenStepExecution.java

```diff
@@ -715,42 +772,79 @@ private String setupGlobalSettingFile(@Nonnull Collection<Credentials> credentia
-            console.format("[withMaven] use Maven global settings provided by the Jenkins global configuration '%s' %n", mvnGlobalSettingsProvider.getSettingsConfigId());
+            if (LOGGER.isLoggable(Level.FINE)) {
+                mavenSettingsLog.append("Config File Provider maven global settings file '").append(mvnGlobalSettingsProvider.getSettingsConfigId()).append("'");
+            }
             globalSettingsFromConfig(mvnGlobalSettingsProvider.getSettingsConfigId(), settingsDest, credentials);
             envOverride.put("GLOBAL_MVN_SETTINGS", settingsDest.getRemote());
+            if (LOGGER.isLoggable(Level.FINE)) {
+                console.println(mavenSettingsLog);
+            }
             return settingsDest.getRemote();
         } else if (globalSettingsProvider instanceof FilePathGlobalSettingsProvider) {
             FilePathGlobalSettingsProvider filePathGlobalSettingsProvider = (FilePathGlobalSettingsProvider) globalSettingsProvider;
             String settingsPath = filePathGlobalSettingsProvider.getPath();
             FilePath settings;
             if ((settings = ws.child(settingsPath)).exists()) {
                 // Global settings file residing on the agent
-                console.format("[withMaven] use Maven global settings provided by the Jenkins global configuration on the build agent '%s' %n", settingsPath);
+                if (LOGGER.isLoggable(Level.FINE)) {
+                    mavenSettingsLog.append("Maven global settings on the build agent '").append(settingsPath).append("'");
+                }
                 settings.copyTo(settingsDest);
+                envOverride.put("GLOBAL_MVN_SETTINGS", settingsDest.getRemote());
+                if (LOGGER.isLoggable(Level.FINE)) {
+                    console.println(mavenSettingsLog);
+                }
+                return settingsDest.getRemote();
             } else {
                 throw new AbortException("Could not find file provided by the Jenkins global configuration '" + settings + "' on the build agent");
             }
-            envOverride.put("GLOBAL_MVN_SETTINGS", settingsDest.getRemote());
-            return settingsDest.getRemote();
         } else if (globalSettingsProvider instanceof DefaultGlobalSettingsProvider) {
             // do nothing
+            if (LOGGER.isLoggable(Level.FINE)) {
+                mavenSettingsLog.append("Maven global settings defined by 'DefaultSettingsProvider', NOT overriding it.");
+                console.println(mavenSettingsLog);
+            }
         } else if (globalSettingsProvider == null) {
             // should not happen according to the source code of jenkins.mvn.GlobalMavenConfig.getGlobalSettingsProvider() in jenkins-core 2.7
             // do nothing
+            if (LOGGER.isLoggable(Level.FINE)) {
+                mavenSettingsLog.append("Maven global settings are null. NO settings will be defined.");
+                console.println(mavenSettingsLog);
+            }

```

Comment:
Also, basically the most of this method's code is repetition of the previous..maybe we could do some otpimisation now that we re there..

Reply:
Not so easy, I tried to do it but gave up. The API for both files is very similar but not the same, we would have to resort to reflection a wrapper or something... so I decided that it is outside of the scope of this change. 

But definitely is repeated code.